[AG-11746] Fix(ssrm): docs fix

sdwvit opened the pull request:
- Include module documentation comment for resetRowHeights

Fix [AG-11746](https://ag-grid.atlassian.net/browse/AG-11746)

sdwvit committed:
Add resetRowHeights method to _RowModelSharedApi

 - Add _RowModelSharedApi import to gridApiFunctions.ts
 - Introduce resetRowHeights method in _RowModelSharedApi interface
 - Include module documentation comment for resetRowHeights

sdwvit commented on packages/ag-grid-community/src/api/gridApiFunctions.ts:
```diff
@@ -291,7 +291,10 @@ export const gridApiFunctionsMap: Record<keyof GridApi, ValidationModuleName> =
         isLastRowIndexKnown: 0,
     }),
-
+    ...mod<_RowModelSharedApi>('RowModelSharedApi', {
```
unsure exactly if I need this, and what is the implication if I do. perhaps @peterjrreynolds knows?

StephenCooper replied:
Not required, as these methods are already listed

sdwvit committed:
Remove onRowHeightChanged and resetRowHeights from _RowModelSharedApi mod

 - Remove _RowModelSharedApi import from gridApiFunctions.ts
 - Eliminate onRowHeightChanged and resetRowHeights from _RowModelSharedApi mod
